packages/ui/src/components/ListV2/Row.tsx

```diff
@@ -0,0 +1,206 @@
+  &[aria-disabled='true'] {
+    pointer-events: none;
+  }
+`
+
+type RowProps = {
+  children: ReactNode
+  id: string
+  expandable?: ReactNode
+  isSelectDisabled?: boolean
+  selectTooltip?: string
+  isDisabled?: boolean
+  sentiment?: (typeof SENTIMENTS)[number]
+}
+
+export const Row = forwardRef(
+  (
+    {
+      children,
+      id,
+      expandable,
+      isDisabled,
+      isSelectDisabled,
+      selectTooltip,
+      sentiment = 'neutral',
+    }: RowProps,
+    ref: ForwardedRef<HTMLDivElement>,
+  ) => {
+    const {
+      allRowSelectValue,
+      areRowSelectable,
+      registerExpandableRow,
+      expandedRowIds,
+      expandRow,
+      collapseRow,
+      registerSelectableRow,
+      selectedRowIds,
+      selectRow,
+      unselectRow,
+    } = useListContext()
+
+    useEffect(() => {
+      if (expandable) {
+        const unregisterCallback = registerExpandableRow(id)
+
+        return unregisterCallback
+      }
+
+      return undefined
+    }, [id, expandable, registerExpandableRow])
+
+    useEffect(() => {
+      if (!isSelectDisabled && !isDisabled) {
+        const unregisterCallback = registerSelectableRow(id)
+
+        return unregisterCallback
+      }
+
+      return undefined
+    }, [id, registerSelectableRow, isSelectDisabled, isDisabled])
+
+    const toggleRowExpand = () => {
+      if (expandedRowIds[id]) {
+        collapseRow(id)
+      } else {
+        expandRow(id)
+      }
+    }
+
+    return (
+      <StyledRow
+        ref={ref}
+        role={!isDisabled && expandable ? 'button row' : 'row'}
+        onClick={!isDisabled && expandable ? toggleRowExpand : undefined}
+        onKeyDown={
+          !isDisabled && expandable
+            ? event => {
+                if (event.key === ' ') {
+                  toggleRowExpand()
+                  event.preventDefault()
+                }
+              }
+            : undefined
+        }
+        tabIndex={!isDisabled && expandable ? 0 : -1}
+        data-sentiment={sentiment}
+        aria-disabled={isDisabled}
+        aria-expanded={expandable ? expandedRowIds[id] : undefined}
+        data-highlight={selectedRowIds[id] === true}
+      >
+        {areRowSelectable ? (
+          <Cell preventClick>
+            <StyledCheckboxContainer
+              data-visibility={
+                allRowSelectValue === false ? 'hover' : undefined
+              }
+            >
+              <Tooltip text={selectTooltip}>
+                <StyledCheckbox
+                  aria-label="select"
+                  checked={selectedRowIds[id]}
+                  value={id}
+                  onChange={() => {
+                    if (selectedRowIds[id]) {
+                      unselectRow(id)
+                    } else {
+                      selectRow(id)
+                    }
+                  }}
+                  disabled={isDisabled || isSelectDisabled}
+                  aria-disabled={isDisabled || isSelectDisabled}
```

Comment:
Here there is an a11y issue, if `disabled` is given you don't need `aria-disabled` it is explicit that the input is disabled